src/retry/index.ts

```diff
@@ -113,11 +113,11 @@ export const retry: RetryMiddleware = ({
                             checkStatus(null, error)
                         })
                     } else {
-                        return Promise.reject(new Error('Number of attempts exceeded.'))
+                        return Promise.reject(error || new Error('Number of attempts exceeded.'))
```

Comment:
If we have a custom error, preserve that custom error.

Reply:
In this particular case, the `until(…)` method should have been called and the developer has explicitely chosen to retry the request by returning `false`. Returning `true` which is also an option would reject the promise with the custom error.

So I think keeping the `Number of attempts exceeded` error label would be better here.

Reply:
Here's my usecase

```js
class ApiError extends Error {
  constructor(message, { code }) {
    super(message)
    this.code = code
  }
}

const api = wretch().middlewares([
  retry({
    until: () => false
  }),
  next => async (url, options) => {
    const res = await next(url, options)
    const body = await res.clone().json()
    if (body.error_message) {
      throw new ApiError(body.error_message, { code: body.error_code })
    }
    return res
  }
])

try {
  await api.get().json()
} catch (err) {
  console.log(err.message, err.code, err instanceof ApiError)
  // Because the err is now `new Error('Number of attempts exceeded.')`
  // You can not access err.code, get original error_message, do the instanceof check anymore.
  // Also you only see a bunch of 'Number of attempts exceeded.' in your server logs, and you have no idea which request failed and why it failed.
}
```

Reply:
Good point, I guess it makes sense in this case.